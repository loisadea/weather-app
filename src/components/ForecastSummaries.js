import React from "react";

import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

function ForecastSummaries({ forecasts, onForecastSelect }) {
  return (
    <div className="ForecastSummaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          temperature={forecast.temperature.max}
          description={forecast.description}
          icon={forecast.icon}
          onSelect={onForecastSelect}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;
