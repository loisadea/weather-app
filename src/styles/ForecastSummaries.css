div.ForecastSummaries {
    display: flex;
}

div.ForecastSummary {
    width: 200px;
    margin: 2px;
    text-align: center;
    background-color: #ccc;
    padding: 20px;
}

div.ForecastSummary__date {
    font-size: 14px;
}

div.ForecastSummary__temperature {
    font-size: 18px;
}

div.ForecastSummary__description {
    font-size: 18px;
}

img.ForecastSummary__icon {
    width: 90px;
}