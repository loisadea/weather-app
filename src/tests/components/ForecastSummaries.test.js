import React from "react";

import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const fakeForecasts = [
    {
      date: 1692197354000,
      temperature: {
        max: 1,
        min: -2,
      },
      description: "Heavy Snow",
      icon: "602",
    },
    {
      date: 1692283754000,
      temperature: {
        max: 20,
        min: 4,
      },
      description: "Tornado",
      icon: "781",
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries forecasts={fakeForecasts} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("has 2 forecasts", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={fakeForecasts} />
    );
    expect(getAllByTestId("ForecastSummary")).toHaveLength(2);
  });
});
