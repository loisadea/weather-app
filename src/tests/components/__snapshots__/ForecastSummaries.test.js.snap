// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ForecastSummaries"
  >
    <div
      class="ForecastSummary"
      data-testid="ForecastSummary"
    >
      <div
        class="ForecastSummary__date"
      >
        Wed, 16 Aug 2023
      </div>
      <img
        alt="Heavy Snow"
        class="ForecastSummary__icon"
        data-testid="SummaryIcon"
        src="https://openweathermap.org/img/wn/13d@4x.png"
      />
      <div
        class="ForecastSummary__temperature"
      >
        1 °C
      </div>
      <div
        class="ForecastSummary__description"
      >
        Heavy Snow
      </div>
      <button
        class="ForecastSummary__more_details"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="ForecastSummary"
      data-testid="ForecastSummary"
    >
      <div
        class="ForecastSummary__date"
      >
        Thu, 17 Aug 2023
      </div>
      <img
        alt="Tornado"
        class="ForecastSummary__icon"
        data-testid="SummaryIcon"
        src="https://openweathermap.org/img/wn/50d@4x.png"
      />
      <div
        class="ForecastSummary__temperature"
      >
        20 °C
      </div>
      <div
        class="ForecastSummary__description"
      >
        Tornado
      </div>
      <button
        class="ForecastSummary__more_details"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
